design/drive-status.md

```diff
@@ -0,0 +1,129 @@
+<!--
+ This work is licensed under a Creative Commons Attribution 3.0
+ Unported License.
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# drive-status
+
+## Status
+
+provisional
+
+## Table of Contents
+
+   * [Title](#title)
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints-optional)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [References](#references)
+
+## Summary
+
+This document explains how drives information, including health data, will be
+retrieved, collected and monitored to provide constant up-to-date status on
+drives present in a server.
+
+## Motivation
+
+We need to be able to collect and update enough information about drives in a
+server to monitor their status and identify them for maintenance and
+replacement.
+Valuable information are for example the current status, type, serial number,
+bay location, part number.
+We need to be able to refresh the information about the status of the drives
+and, if a drive has been replaced, we need to be able to retrieve the
+new information and update the hardware inventory to provide a constant
+up-to-date vision on the status of the drives.
+
+### Goals
+
+1. Decide which information on the drives to retrieve
+2. Agree on how to retrieve such information
+3. Provide a way to manually/automatically update the information
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+### Implementation Details/Notes/Constraints
+
+- For each drive we need to collect at least the following information:
+    * Capacity
+    * Location
+    * Manufacturer
+    * Model
+    * Part Number
+    * Serial Number
+    * Status
+    * SMART details
+    
+- NVMe drives require special tools to get SMART data, although
+[smartmontools](https://www.smartmontools.org/wiki/NVMe_Support) supports NVMe
+devices since version 6.5
```

Comment:
Do we need to update python-hardware and/or IPA to support it?

Reply:
What sort of information can we get on libvirt VMs? We're still primarily testing with VMs

Reply:
@dtantsur: python-hardware as it is now basically ignores nvme devices:
https://github.com/redhat-cip/hardware/blob/master/hardware/diskinfo.py#L33
I tested it and the disk are not reported by the hardware-detect command.
One solution could be to modify that module and maybe even add support for nvme tools (adding a new parser in this case).

@dhellmann: if you mean running smartmontools on vd* devices, I don't think that's possible as you can only use smartmontools on actual physical drives.
But maybe I got this wrong?

Reply:
I went ahead and started modifying python-hardware to get info on nvme disks, getting some first results on my laptop:
```
[('disk', 'logical', 'count', '1'),
 ('disk', 'nvme0n1', 'size', '512'),
 ('disk', 'nvme0n1', 'model', 'SAMSUNG MZVLB512HAJQ-000L7'),
 ('disk', 'nvme0n1', 'optimal_io_size', '0'),
 ('disk', 'nvme0n1', 'physical_block_size', '512'),
 ('disk', 'nvme0n1', 'rotational', '0'),
 ('disk', 'nvme0n1', 'nr_requests', '1023'),
 ('disk', 'nvme0n1', 'scheduler', 'none'),
 ('disk', 'nvme0n1', 'id', 'nvme-eui.0025388981b211a2'),
 ('disk', 'nvme0n1', 'id', 'nvme-SAMSUNG_MZVLB512HAJQ-000L7_S3TNNF0K939750'),
```